test/unit/io/sequence_file/sequence_file_format_sam_test.cpp

```diff
@@ -73,74 +73,48 @@ struct read : public ::testing::Test
     }
 };
 
-TEST_F(read, standard)
+TEST_F(read_sam, standard)
 {
     std::string input
     {
-R"(@ Comment
-@ Blablabla
-@ Bla   bla bla
-ID1	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-
+R"(ID1	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-
 ID2	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT	!##$&'()*+,-./+)*+,-)*+,-)*+,-)*+,BDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDE
 ID3 lala	0	*	0	0	*	*	0	0	ACGTTTA	!!!!!!!
 )"
     };
     do_read_test(input);
 }
 
-TEST_F(read, tags)
+TEST_F(read_sam, tags)
 {
     std::string input
     {
-R"(@ Comment
-@ Blablabla
-@ Bla   bla bla
-ID1	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-	FI:i:1
+R"(ID1	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-	FI:i:1
 ID2	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT	!##$&'()*+,-./+)*+,-)*+,-)*+,-)*+,BDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDE	AS:i:3
 ID3 lala	0	*	0	0	*	*	0	0	ACGTTTA	!!!!!!!	TI:i:2
 )"
     };
     do_read_test(input);
 }
 
-TEST_F(read, mixed_issues)
+TEST_F(read_sam, mixed_issues)
 {
   std::string input
   {
-R"(@ Comment
-ID1	0	BABABA	200	0	*	BABABA	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-	FI:i:1
+R"(ID1	0	BABABA	200	0	*	BABABA	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-	FI:i:1
 ID2	0	*	0	0	BABA	*	30	0	ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT	!##$&'()*+,-./+)*+,-)*+,-)*+,-)*+,BDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDE
 ID3 lala	0	*	0	0	HAHAHAHA+	*	0	0	ACGTTTA	!!!!!!!
 )"
     };
     do_read_test(input);
 }
 
-TEST_F(read, options_truncate_ids)
-{
-  std::string input
-  {
-R"(@ Comment
-ID1 b	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTT	!##$%&'()*+,-./++-
-ID2 lala2	0	*	0	0	*	*	0	0	ACGTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT	!##$&'()*+,-./+)*+,-)*+,-)*+,-)*+,BDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDEBDE
-ID3 lala	0	*	0	0	*	*	0	0	ACGTTTA	!!!!!!!
-)"
-    };
-
-    options.truncate_ids = true;
-    expected_ids[2] = "ID3"; // "lala" is stripped
-    do_read_test(input);
-}
-
-TEST_F(read, seq_qual)
+TEST_F(read_sam, seq_qual)
```

Comment:
You deleted a test? :thinking:

Reply:
Yes, the one that tests if a SAM read `id` is truncated... which decided is not the behaviour we want in a TAB separated file. This is FASTA/Q specific. It also does not apply to EMBL and Genbank. So I thought it is fine to not apply to SAM either?

Reply:
The thing is that independent of the concrete format, usually the important part of the ID is before the first whitespace. For protein sequences IDs can be very long (much longer than the sequence) and often you want to discard the stuff after first whitespace.

Then again, SAM cannot hold protein sequences.

Reply:
> Then again, SAM cannot hold protein sequences

That's why I thought this adds overhead code that is probably never needed. We can add this easily later if requested as a feature.